### ISSUE TITLE
[Bug?] obj mesh z direction got flipped.

### ISSUE THREAD

**Issue by adfriz** [leezer3/OpenBVE#866]
### Description

when loading obj mesh in object viewer the mesh z direction got flipped from -Z forward to +Z forward.

using assimp obj parser because original obj parser give an error.

### Reproduction

here the object :
[test.zip](https://github.com/leezer3/OpenBVE/files/10728467/test.zip)

1.obj is exported from blender 2.79 cannot loaded and give an error index out of range.
2.obj is exported from blender 3.4.1 using legacy obj exporter, loaded successfully with assimp parser but direction flipped.
3.obj is exported from blender 3.4.1 using new obj exporter, loaded successfully with assimp parser but direction flipped.

## Route

no route

## Train

no train

## Logs

no logs but 1.obj give an index out of range error in object viewer.

### Related information 

* Operating system:  windows 10 21h2
* Method of control : keyboard, mouse
* OpenBVE : 1.8.4.4 installer version

**Comment by leezer3** [comment]
The reversed Z is not (technically) a bug.
OBJs can be either way [DirectX or OpenGL] depending on the exporter.
If it contains a recognisable exporter header, I can add it to the list of those needing mirroring.

Will look at the out of range issue tomorrow maybe.

**Comment by adfriz** [comment]
The header only display name what program that obj from.
The exporter also have an option to choose forward axis.

I just dont understand why the obj get flipped even the default axis for the exporter is +Z forward and Y+ Up 🤣, because if i remember several version before the obj is not flipped like now.
I think i will choose -Z forward as default for now.

Honestly i only use obj file in openbve for personal test because the exporter very fast, for final addons i only use b3d or csv because AFAIK is hard to convert to other format if the user dont know how to do it.

**Comment by leezer3** [comment]
TLDR:
A while back, we hadn't realised that objs could be like this & there was a bunch of special casing going on....
Assimp was introduced and we realised the root issue was crummy format specification.
The aim is for both importers to be consistant.

**Comment by leezer3** [comment]
Fixed the Assimp bug.

The original obj parser fails to load all of these at the minute. Looking into that next....

**Comment by leezer3** [comment]
OK, try the latest build (will be there in a few minutes)

I think everything is now right with both parsers.
A little concerned that I seem to have inverted the dissolve factor in the original parser (which suggests something in my original test set used it this way), but we'll see how it goes.

With regards to the co-ordinate detection, unfortunately if Blender doesn't write *which* co-ordinate system it's using in the exported file & offers a choice, not a lot we can do.
I'll file a bug at the blender end to see if they want to add this to the exported files (allowing us to auto-handle said objects)
